django_project/property/signals.py

```diff
@@ -0,0 +1,36 @@
+import json
+from django.core.management import call_command
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from frontend.models.context_layer import ContextLayer
+from frontend.models.spatial import SpatialDataModel, SpatialDataValueModel
+from .models import Property
+
+
+@receiver(post_save, sender=Property)
+def handle_spatial_data(sender, instance, created, **options):
+    if created:
+        context_layers = ContextLayer.objects.all()
+
+        for context_layer in context_layers:
+            # Create spatial data instance
+            spatial_data_instance = SpatialDataModel(
+                property=instance,
+                context_layer=ContextLayer.objects.get(id=context_layer.id)
```

Comment:
Why do you query the context layer again here? Can you just use the context_layer.id?